### ISSUE TITLE
Map function runs serial when forced to parallel

### ISSUE THREAD

**Issue by wehlgrundspitze** [hyperspy/hyperspy#3152]
#### Describe the bug
Despite indicating to the map function to perform the operation in parallel, it is executed in series and takes the time expected for serial execution.

#### To Reproduce
Minimum working example for parallel mapping as written in the hyperspy documentation:

```
import os
import time

import hyperspy.api as hs
import numpy as np

def slow_func(data):
    print("Func exec")
    time.sleep(1.)

    return data + 1

print(os.cpu_count())

s = hs.signals.Signal1D(np.arange(40).reshape((20, 2)))
print(s)

s.map(slow_func, parallel=True, max_workers=4)
```

#### Expected behavior
4 workers should run in parallel and the map function should finish in 5s. Instead it takes the 20s expected for serial operation.

#### Python environement:
 - HyperSpy version: 1.7.5
 - Python version: 3.10.6

#### Additional context

Since the parallel operation is performed internally via the dask framework, I went through the basic dask tutorial at https://tutorial.dask.org/02_array.html. The parallel operations within this tutorial were performed correctly with shorter execution time.

I found this behaivior on two different machines with Ubuntu 22.04. The phenomenon occurred while running the code in a Jupyter Nottebook as well es execute it with the c-python interpreter.

**Comment by wehlgrundspitze** [comment]
Just for completeness: The print statement about the number of processors shows correctly 6 respectively 8 on the two machines.

**Comment by CSSFrancis** [comment]
Admittedly this bug results from how we use dask to run things in parallel now. You need at least 4 chunks to run on 4 cores so we should be creating a dask array with 4 chunks.

If you make a bigger array it should work. And if you make a lazy array you can define the number of chunks correctly.

**Comment by CSSFrancis** [comment]
For example this should work

```python
import os
import time

import hyperspy.api as hs
import numpy as np


def slow_func(data):
    print("Func exec")
    time.sleep(1.)

    return data + 1

print(os.cpu_count())

s = hs.signals.Signal1D(np.arange(40).reshape((20, 2)))
s=s.as_lazy()
s.rechunk(5)
print(s)

s.map(slow_func, parallel=True, max_workers=4, lazy_output=False)
```

**Comment by CSSFrancis** [comment]
The issue is here:

Mostly just the first line where the signal is cast to a lazy signal.

https://github.com/hyperspy/hyperspy/blob/70ca0762a332a7239a4c1eb75001062ab6c7495d/hyperspy/signal.py#L4964-L4988


Something like:

```python
if not self._lazy:
    s_input = self.as_lazy()
    nav_shape = s_input.axes_manager.navigation_shape()
    nav_chunks =  # ... split the nav_shape into equal chunks
    s_input.rechunk(nav_chunks)
```
Ideally we want the number of chunks to be at least ~3 times the number of workers requested but more isn't a big deal. I'm away from my computer this weekend but I can try to get around to this Monday.  Otherwise if @wehlgrundspitze if you want to take a crack at it I am happy to help!

Better yet the `as_lazy` function should take a `chunks` kwarg similar to the rechunk function for lazy signals:

https://github.com/hyperspy/hyperspy/blob/70ca0762a332a7239a4c1eb75001062ab6c7495d/hyperspy/_signals/lazy.py#L274-L279

**Comment by wehlgrundspitze** [comment]
Thanks for this comprising answers. I will have a look how to set the chunks correctly and make a pull request.

**Comment by CSSFrancis** [comment]
Closed by #3198
